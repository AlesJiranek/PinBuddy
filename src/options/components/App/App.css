@keyframes slideIn {
  0% {
    opacity: 0;
    transform: translateX(calc(var(--lh) / 4));
  }

  100% {
    opacity: 1;
    transform: none;
  }
}

.app {
  display: flex;
  flex-direction: column;
  width: 100%;
  max-width: calc(var(--lh) * 22);
  margin: 0 auto;
  padding: var(--lh);
  min-height: 100vh;
}

.app__header,
.app__main,
.app__footer {
  animation-name: slideIn;
  animation-duration: var(--animation-duration);
  animation-fill-mode: backwards;
}

.app__header {
  animation-delay: 100ms;
}

.app__main {
  animation-delay: 150ms;
}

.app__footer {
  animation-delay: 200ms;
  margin-top: auto;
}
