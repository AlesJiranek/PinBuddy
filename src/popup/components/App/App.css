.app {
  width: calc(var(--lh) * 22);
  height: calc(var(--lh) * 20);
  overflow: auto;
  padding: var(--lh);
}

.app__header {
  display: flex;
  justify-content: space-between;
}

.app__logo {
  display: flex;
}

.app__main {
}

