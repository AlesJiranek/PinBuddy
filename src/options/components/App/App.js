import React, { Component } from 'react';
import './App.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Login from '../Login';
import { updateUserInfo } from '../../actions/auth';
import { checkConnection, wentOffline, wentOnline } from '../../actions/online';
import Footer from '../Footer';
import Options from '../Options';
import { Title, Paragraph } from 'theme';

class App extends Component {
  componentDidMount() {
    window.addEventListener('online', this.handleOnlineEvent);
    window.addEventListener('offline', this.handleOfflineEvent);

    this.props.checkConnection(navigator.onLine);

    chrome.storage.local.get(['username'], result => {
      if (result.username) {
        this.props.updateUserInfo(result);
      }
    });
  }

  componentWillUnmount() {
    window.removeEventListener('online', this.handleOnlineEvent);
    window.removeEventListener('offline', this.handleOfflineEvent);
  }

  render() {
    return (
      <div className="app">

        <header className="app__header">
          <Title t={chrome.i18n.getMessage('optionsTitle')} />
        </header>

        <main className="app__main">
          {
            !this.props.online ?
              <Paragraph t={chrome.i18n.getMessage('optionsOfflineMessage')} /> :
              this.renderOnline()
          }
        </main>

        <footer className="app__footer">
          <Footer t={chrome.i18n.getMessage('optionsFooter')} />
        </footer>

      </div>
    );
  }

  renderOnline() {
    return this.props.username ? <Options /> : <Login />;
  }

  handleOnlineEvent = () => {
    this.props.wentOnline();
  }

  handleOfflineEvent = () => {
    this.props.wentOffline();
  }
}

App.propTypes = {
  username: PropTypes.string.isRequired,
  online: PropTypes.bool.isRequired,
  updateUserInfo: PropTypes.func.isRequired,
  checkConnection: PropTypes.func.isRequired,
  wentOnline: PropTypes.func.isRequired,
  wentOffline: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return {
    username: state.auth.username,
    online: state.online,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    updateUserInfo: userInfo => dispatch(updateUserInfo(userInfo)),
    checkConnection: online => dispatch(checkConnection(online)),
    wentOffline: () => dispatch(wentOffline()),
    wentOnline: () => dispatch(wentOnline()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
