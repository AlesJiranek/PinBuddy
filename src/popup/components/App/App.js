import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getUserInfo } from '../../actions/user';
import { checkConnection, wentOffline, wentOnline } from '../../actions/online';
import Logo from '../Logo';
import Nav from '../Nav';
import Main from '../Main';
import './App.css';

class App extends Component {
  componentDidMount() {
    window.addEventListener('online', this.handleOnlineEvent);
    window.addEventListener('offline', this.handleOfflineEvent);
    this.props.checkConnection(navigator.onLine);
    this.props.getUserInfo();
  }

  componentWillUnmount() {
    window.removeEventListener('online', this.handleOnlineEvent);
    window.removeEventListener('offline', this.handleOfflineEvent);
  }

  render() {
    return (
      <div className="app">
        <header className="app__header">
          <div className="app__logo">
            <Logo />
          </div>
          <nav className="app__logo">
            <Nav />
          </nav>
        </header>
        <hr />
        <main className="app__main">
          <Main />
        </main>
      </div>
    );
  }

  handleOnlineEvent = () => {
    this.props.wentOnline();
  }

  handleOfflineEvent = () => {
    this.props.wentOffline();
  }
}

App.propTypes = {
  getUserInfo: PropTypes.func.isRequired,
  checkConnection: PropTypes.func.isRequired,
  wentOnline: PropTypes.func.isRequired,
  wentOffline: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => {
  return {
    getUserInfo: () => dispatch(getUserInfo()),
    checkConnection: online => dispatch(checkConnection(online)),
    wentOnline: () => dispatch(wentOnline()),
    wentOffline: () => dispatch(wentOffline()),
  };
};


export default connect(null, mapDispatchToProps)(App);
