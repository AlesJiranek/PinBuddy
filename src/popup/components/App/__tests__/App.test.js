import React from "react";
import { shallow } from "enzyme";

import { App } from "../App";

describe("<App />", () => {
  const defaultProps = {
    getUserInfo: jest.fn(),
    checkConnection: jest.fn(),
    wentOnline: jest.fn(),
    wentOffline: jest.fn()
  };

  it("should render correctly", () => {
    const tree = shallow(<App {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });
});
