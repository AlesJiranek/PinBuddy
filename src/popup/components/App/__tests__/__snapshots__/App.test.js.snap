// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    checkConnection={
      [MockFunction] {
        "calls": Array [
          Array [
            true,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getUserInfo={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    wentOffline={[MockFunction]}
    wentOnline={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="app__header"
        >
          <div
            className="app__logo"
          >
            <Connect(Logo) />
          </div>
          <nav>
            <Connect(Nav) />
          </nav>
        </header>,
        <main
          className="app__main"
        >
          <Connect(Main) />
        </main>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="app__logo"
            >
              <Connect(Logo) />
            </div>,
            <nav>
              <Connect(Nav) />
            </nav>,
          ],
          "className": "app__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Logo) />,
              "className": "app__logo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Nav) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "nav",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Main) />,
          "className": "app__main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="app__header"
          >
            <div
              className="app__logo"
            >
              <Connect(Logo) />
            </div>
            <nav>
              <Connect(Nav) />
            </nav>
          </header>,
          <main
            className="app__main"
          >
            <Connect(Main) />
          </main>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="app__logo"
              >
                <Connect(Logo) />
              </div>,
              <nav>
                <Connect(Nav) />
              </nav>,
            ],
            "className": "app__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Logo) />,
                "className": "app__logo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Nav) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "nav",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Main) />,
            "className": "app__main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
